app/models/course/achievement.rb

```diff
@@ -8,7 +8,11 @@ class Course::Achievement < ActiveRecord::Base
   belongs_to :course, inverse_of: :achievements
   has_many :course_user_achievements, class_name: Course::UserAchievement.name,
                                       inverse_of: :achievement, dependent: :destroy
-  has_many :course_users, through: :course_user_achievements, class_name: CourseUser.name
+  # Due to the through relationship, destroy dependent had to be added for course users in order for
+  # UserAchievement's destroy callbacks to be called, However, this destroy dependent will not
+  # actually remove the course users when the Achievement object is destroyed.
```

Comment:
`If using with the :through option, the association on the join model must be a belongs_to, and the records which get deleted are the join records, rather than the associated records.`
http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html

However, I could not find any new report on this issues as well. I will see if there is any alternative way to work on this if this doesn't seem suitable. Any thoughts on it?